k2/csrc/intersect_pruned.cu

```diff
@@ -0,0 +1,1044 @@
+  os << "ArcInfo{" << a.a_fsas_arc_idx012 << "," << a.arc_loglike << ","
+     << a.u.dest_a_fsas_state_idx01 << "," << a.end_loglike << "}";
+  return os;
+}
+*/
+
+}  // namespace intersect_pruned_internal
+
+using namespace intersect_pruned_internal;  // NOLINT
+
+// Caution: this is really a .cu file.  It contains mixed host and device code.
+
+/*
+   Pruned intersection (a.k.a. composition) that corresponds to decoding for
+   speech recognition-type tasks.  Can use either different decoding graphs (one
+   per acoustic sequence) or a shared graph
+*/
+class MultiGraphDenseIntersectPruned {
+ public:
+  /**
+     Pruned intersection (a.k.a. composition) that corresponds to decoding for
+     speech recognition-type tasks
+
+       @param [in] a_fsas  The decoding graphs, one per sequence.  E.g. might
+                           just be a linear sequence of phones, or might be
+                           something more complicated.  Must have either the
+                           same Dim0() as b_fsas, or Size0()==1 in which
+                           case the graph is shared.
+       @param [in] b_fsas  The neural-net output, with each frame containing the
+                           log-likes of each phone.  A series of sequences of
+                           (in general) different length.
+       @param [in] search_beam    "Default" search/decoding beam.  The actual
+                           beam is dynamic and also depends on max_active and
+                           min_active.
+       @param [in] output_beam    Beam for pruning the output FSA, will typically
+                           be smaller than search_beam.
+       @param [in] min_active  Minimum number of FSA states that are allowed to
+                           be active on any given frame for any given
+                           intersection/composition task. This is advisory,
+                           in that it will try not to have fewer than this
+                           number active.
+       @param [in] max_active  Maximum number of FSA states that are allowed to
+                           be active on any given frame for any given
+                           intersection/composition task. This is advisory,
+                           in that it will try not to exceed that but may not
+                           always succeed.
+   */
+  MultiGraphDenseIntersectPruned(FsaVec &a_fsas, DenseFsaVec &b_fsas,
+                                 float search_beam, float output_beam,
+                                 int32_t min_active, int32_t max_active)
+      : a_fsas_(a_fsas),
+        b_fsas_(b_fsas),
+        search_beam_(search_beam),
+        output_beam_(output_beam),
+        min_active_(min_active),
+        max_active_(max_active),
+        dynamic_beams_(a_fsas.Context(), b_fsas.shape.Dim0(), search_beam) {
+    NVTX_RANGE(__func__);
+    c_ = GetContext(a_fsas.shape, b_fsas.shape);
+    K2_CHECK(b_fsas.scores.IsContiguous());
```

Comment:
Does this constraint still hold? I find that the current code
is using `Accessor` to access its element.